pkgs/tools/backup/gphotos-sync/default.nix

```diff
@@ -0,0 +1,44 @@
+{ lib, fetchFromGitHub, python3Packages }:
+
+python3Packages.buildPythonApplication rec {
+  pname = "gphotos-sync";
+  version = "2.14.2";
+
+  src = fetchFromGitHub {
+    owner = "gilesknap";
+    repo = "gphotos-sync";
+    rev = version;
+    sha256 = "0cfmbrdy6w18hb623rjn0a4hnn3n63jw2jlmgn4a2k1sjqhpx3bf";
+  };
+
+  propagatedBuildInputs = with python3Packages; [
+    appdirs
+    attrs
+    exif
+    psutil
+    pyyaml
+    requests_oauthlib
+  ];
+  checkInputs = with python3Packages; [
+    pytestCheckHook
+    mock
+  ];
+  checkPhase = ''
+    export HOME=$(mktemp -d)
+
+    # patch to skip all tests that do network access
```

Comment:
Inspired by some other tests in this repo, this is a simple way to skip any test that does network access (which is 24/34 of them in this repo). Is this sort of thing acceptable here? Unfortunately the tests that do network access aren't confined to a particular directory, so they're hard to skip by name.

(The tests download actual data from an actual google photos account.)

Reply:
Sure, we could list all 24 failing tests, but that seems like it would be less maintainable: the list would have to be updated if new tests were added. This detects them all accurately and automatically.